### PR TITLE
[BACKLOG-18166] Intermediary Change to run work items in a background…

### PR DESCRIPTION
**Pull request by pentaho-esager**
… thread which waits for the pentaho server to be completely available before launching a work item request

### CODE REVIEW

**Comment by pentaho-esager** [comment]
@pentaho/rogueone

File: extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java [hunk 1]
```diff
@@ -77,14 +90,76 @@ private void placeBreadcrumbFile( String folderPath, boolean success ) {
...
+    int timeout = 600000;
+    final int sleep_time = 5000;
+    boolean isAlive = false;
+    while ( !isAlive ) {
+      int statusCode = pingIsAlive();
+      if ( statusCode == HttpStatus.SC_OK ) {
+        isAlive = true;
+      } else {
+        if ( timeout == 0 ) {
+          placeBreadcrumbFile( getSolutionPath(), false );
```
Comment (neeta-rathi) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java]: So we don't exit the while loop, but write the breadcrumb instead and that eventually leads to exiting the container, correct?

File: extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java [hunk 2]
```diff
@@ -77,14 +90,76 @@ private void placeBreadcrumbFile( String folderPath, boolean success ) {
...
+      final HttpResponse response = client.execute( pingIsAlive );
+      return response.getStatusLine().getStatusCode();
+    } catch ( URISyntaxException | IOException e ) {
+      logger.error( e.getMessage() );
+    }
+    return HttpStatus.SC_INTERNAL_SERVER_ERROR;
+  }
+
+  private void waitUntilServerIsAlive( ) {
+    int timeout = 600000;
```
Comment (pdesai16) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java]: same as above. Adding a debug statement would help in troubleshooting later

File: extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java (hunk 1 above)
Comment (pentaho-esager) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java]: Yes, that was my intent, but we can also break if we want.

File: extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java [hunk 3]
```diff
@@ -77,14 +90,76 @@ private void placeBreadcrumbFile( String folderPath, boolean success ) {
       logger.error( Messages.getInstance().getErrorString( "ActionInvokerSystemListener.ERROR_0004_COULD_NOT_WRITE_STATUS" ) );
     }
   }
+
+  private int pingIsAlive( ) {
+    try {
```
Comment (pentaho-esager) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java]: Sure

File: extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java (hunk 1 above)
Comment (neeta-rathi) [inline extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java]: This is fine.

**Comment by pentaho-esager** [comment]
Updated with requested changes

**Pull request merged**